### PR TITLE
Introduce Java 9 module compatibility

### PR DESCRIPTION
**Pull request by trustin**
Motivation:

Some users are having issues with using Armeria as a Java 9 module because:

- Armeria core JAR contains `module-info.class` from Bouncy Castle, which should be removed during the shading process.
- Armeria doesn't define module metadata at all

Modifications:

- Updated `gradle-scripts` to exclude `module-info.class` while shading.
  - This removes Bouncy Castle's `module-info.class` from the final JAR, fixing the issue mentioned in the previous section.
- Updated `gradle-scripts` to auto-inject the `Automatic-Module-Name` property into `META-INF/MANIFEST.MF` so that JVM auto-generates the module metadata.
- Miscellaneous:
  - Overwrote the `Created-By` property in `META-INF/MANIFEST.MF` so that the manifest file's content doesn't change depending on JDK version.

Result:

- (new feature) Armeria JARs are now Java modules.

### CODE REVIEW

File: gradle/scripts/lib/java.gradle [hunk 1]
```diff
@@ -154,6 +161,15 @@ configure(projectsWithFlags('java')) {
         options.compilerArgs += '-parameters'
     }
 
+    // Set the 'Automatic-Module-Name' property in 'MANIFEST.MF' if `automaticModuleNames` is true.
```
Comment (minwoox) [inline gradle/scripts/lib/java.gradle:164]:
```suggestion
    // Set the 'Automatic-Module-Name' property in 'MANIFEST.MF' if `automaticModuleName` is not null.
```

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by ikhoon — APPROVED** [review]
The changes look good!

**Review by jrhee17 — APPROVED** [review]
Checked that `Automatic-Module-Name` is set for output jars 👍   Thanks @trustin 🙇 👍 🙇

**Pull request merged**